### PR TITLE
pagmo 2.9 (new formula)

### PR DESCRIPTION
**Pull request by jslee02**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/pagmo.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+          // 4 - Run the evolution in parallel on the 16 separate islands 10 times.
+          archi.evolve(10);
+
+          // 5 - Wait for the evolutions to be finished
+          archi.wait_check();
+
+          // 6 - Print the fitness of the best solution in each island
+          for (const auto &isl : archi) {
+              std::cout << isl.get_population().champion_f()[0] << std::endl;
+          }
```
Comment (lembacon) [inline Formula/pagmo.rb]: We may want to add a `return 0;` in order to suppress the compiler warning.

File: Formula/pagmo.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      }
+    EOS
+    system ENV.cxx, "test.cpp", "-I#{Formula["eigen"].include}/eigen3",
+                    "-I#{include}", "-std=c++11", "-o", "test"
+    system "./test"
```
Comment (lembacon) [inline Formula/pagmo.rb]: It might worth verifying the output with `shell_output` and `assert_match`.

Comment (lembacon) [reply Formula/pagmo.rb]: If the result is not deterministic, then it should be fine to leave it as-is. 👍

**Review by lembacon — APPROVED** [review]

**Comment by lembacon** [comment]
Shipped 🚢. Thanks for your contribution @jslee02!

**Comment by jslee02** [comment]
Thanks for the prompt review and merge! 👍

**Pull request closed without merging**